google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PublisherStats.java

```diff
@@ -16,44 +16,24 @@
 
 package com.google.cloud.pubsub;
 
-import javax.annotation.concurrent.Immutable;
+import com.google.auto.value.AutoValue;
 
 /**
  * A snapshot of the publisher statistics at the time they were requested from the {@link
  * Publisher}.
  */
 //TODO: Finish implementation.
-@Immutable
-public class PublisherStats {
-  private final long sentMessages;
-  private final long ackedMessages;
-  private final long failedMessages;
-  private final long pendingMessages;
-
-  PublisherStats(long sentMessages, long ackedMessages, long failedMessages, long pendingMessages) {
-    this.sentMessages = sentMessages;
-    this.ackedMessages = ackedMessages;
-    this.failedMessages = failedMessages;
-    this.pendingMessages = pendingMessages;
-  }
-
+@AutoValue
+public abstract class PublisherStats {
```

Comment:
@eamonnmcmanus I thought of just getting it to compile then work on the rest later. I guess the work is small enough that it makes sense to roll in.